stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/viewmodel/BaseMessageListHeaderViewModel.kt

```diff
@@ -1,70 +1,95 @@
-    private val _members = MediatorLiveData<List<Member>>()
-    private val _channelState = MediatorLiveData<Channel>()
-    private val _anyOtherUsersOnline = MediatorLiveData<Boolean>()
-    private val _typingUsers = MediatorLiveData<List<User>>()
-
-    public val activeThread: LiveData<Message?> = _activeThread
-    public val members: LiveData<List<Member>> = _members
-    public val channelState: LiveData<Channel> = _channelState
-    public val anyOtherUsersOnline: LiveData<Boolean> = _anyOtherUsersOnline
-    public val online: LiveData<ConnectionState> = chatDomain.connectionState
-    public val typingUsers: LiveData<List<User>> = _typingUsers
+    /**
+     * Holds information about the current channel and is actively updated.
+     */
+    public val channelState: ChannelState =
+        chatClient.asReferenced().watchChannel(cid, MessageListViewModel.MESSAGES_LIMIT).asState(viewModelScope)
 
-    private val logger = ChatLogger.get("MessageListHeaderViewModel")
+    /**
+     * The current [Channel] created from [ChannelState]. It emits new data either when
+     * channel data or the list of members in [ChannelState] updates.
+     *
+     * Combining the two is important because members changing online status does not result in
+     * channel events being received.
+     */
+    public val channel: LiveData<Channel> =
+        Transformations.map(channelState.channelData.combine(channelState.members) { _, _ -> }.asLiveData()) {
+            channelState.toChannel()
+        }
 
-    init {
-        chatDomain.watchChannel(cid, 0).enqueue { channelControllerResult ->
-            if (channelControllerResult.isSuccess) {
-                val channelController = channelControllerResult.data()
-                _members.addSource(channelController.members) { _members.value = it }
-                _channelState.addSource(map(channelController.offlineChannelData) { channelController.toChannel() }) {
-                    _channelState.value = it
-                }
-                _channelState.addSource(map(channelController.members) { channelController.toChannel() }) {
-                    _channelState.value = it
-                }
-                _anyOtherUsersOnline.addSource(
-                    map(channelController.members) { members ->
-                        members.asSequence()
-                            .filter { it.user != chatDomain.user.value }
-                            .any { it.user.online }
-                    }
-                ) { _anyOtherUsersOnline.value = it }
-                _typingUsers.addSource(channelController.typing) { typingEvent ->
-                    _typingUsers.value = typingEvent.users
-                }
-            } else {
-                logger.logE("Could not watch channel with cid: $cid. Error: ${channelControllerResult.error()}")
-            }
+    /**
+     * A list of users who are currently typing.
+     */
+    public val typingUsers: LiveData<List<User>> =
```

Comment:
Should we expose this `LiveData` if it can be already accessed through `channelState`? 🤔

Reply:
Matter resolved in a meeting.